### ISSUE TITLE
`iter zip-into-record` returns a table

### ISSUE THREAD

**Issue by NotTheDr01ds** [nushell/nushell#14062]
### Describe the bug

`iter zip-into-record` seems ... at the least misnamed, but potentially just wrong.

History: #9380 and #9395
cc: @amtoine and @AstrickHarren 

### How to reproduce

```nu
use std/iter
[ a b c ] | iter zip-into-record [ foo bar baz ]

# => ╭───┬─────┬─────┬─────╮
# => │ # │  a  │  b  │  c  │
# => ├───┼─────┼─────┼─────┤
# => │ 0 │ foo │ bar │ baz │
# => ╰───┴─────┴─────┴─────╯

[ a b c ] | iter zip-into-record [ foo bar baz ] | describe

# => table<a: string, b: string, c: string>
```

### Expected behavior

Given the name of the command, I would expect:

```nu
╭───┬─────╮
│ a │ foo │
│ b │ bar │
│ c │ baz │
╰───┴─────╯
```

Note that this is the result of (today):

```nu
[ a b c ] | iter zip-into-record [ foo bar baz ] | transpose | transpose -dr 
```

Should the command be renamed or changed?

### Configuration

| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.98.1                                   |
| major              | 0                                        |
| minor              | 98                                       |
| patch              | 1                                        |
| branch             | virtual-mods                             |
| commit_hash        | 1dc58b84a85a7d7e8fa4a9fec866582eddcfc036 |
| build_os           | linux-x86_64                             |
| build_target       | x86_64-unknown-linux-gnu                 |
| rust_version       | rustc 1.79.0 (129f3b996 2024-06-10)      |
| rust_channel       | 1.79.0-x86_64-unknown-linux-gnu          |
| cargo_version      | cargo 1.79.0 (ffa9cf99a 2024-06-03)      |
| build_time         | 2024-10-09 13:39:29 -04:00               |
| build_rust_channel | release                                  |
| allocator          | mimalloc                                 |
| features           | default, sqlite, trash                   |
| installed_plugins  |                                          |

**Comment by NotTheDr01ds** [comment]
I might be wrong, but given the output, this seems like it should be written:

```nu
[ foo bar baz ] | to-columns [ a b c ]
```

**Comment by amtoine** [comment]
yeah i used it in the past and thought the naming was quite strange for what it's doing :eyes:
